backend/service-quiz/src/main/scala/pl/smtc/smartwords/client/WordService.scala

```diff
@@ -38,20 +38,31 @@ class WordService {
 
   /**
    * Method used to communicate with word service and retrieve a random word
+   * @param language type of language for the word to be retrieved
+   * @throws WordServiceException when the response from word service is invalid
    * @return random word object
    */
-  def getRandomWord: Word = {
-    sendGetWordsRequest(wordsEndpoint.withQueryParam("size", "1")
-                                     .withQueryParam("random", "true")).head
+  @throws(classOf[WordServiceException])
+  def getRandomWord(language: String): Word = {
+    val endpoint: Uri = wordsEndpoint.addSegment(language)
+                                     .withQueryParam("size", "1")
+                                     .withQueryParam("random", "true")
+    val receivedWord: List[Word] = sendGetWordsRequest(endpoint)
+    if (receivedWord.isEmpty) {
+      throw new WordServiceException("Invalid input parameter(s).")
+    }
+    receivedWord.head
   }
 
   /**
    * Method used to communicate with word service and retrieve all words with specified category
+   * @param language of the word which we want to retrieve
    * @param category type of words category to be retrieved
    * @return list of all words with specified category
    */
-  def getWordsByCategory(category: String): List[Word] = {
-    sendGetWordsRequest(wordsEndpoint.withQueryParam("cat", category))
+  def getWordsByCategory(language: String, category: String): List[Word] = {
+    sendGetWordsRequest(wordsEndpoint.addSegment(language)
+                                     .withQueryParam("cat", category))
```

Comment:
We should also check if returned value is non-empty (just like in `getRandomWord` above)

Reply:
Updated.